app/controllers/application_controller.rb

```diff
@@ -18,8 +20,64 @@ def set_checkout_redirect
     end
   end
 
+  def enable_embedded_styles
+    session[:embedded_shopfront] = true
+    render json: {}, status: 200
```

Comment:
So if someone figures out they can do an ajax request to a non-public path and change the CSS a little bit it's basically ok...